adam_preprocessing/shape_stroke_extraction.py

```diff
@@ -510,6 +544,35 @@ def get_strokes(self):
                     sub_part=None,
                 )
             )
+
+        # Calculate relative sizes using already-calculated absolute sizes:
+        for i in range(len(data)):
+            data[i]['relative_size'] = {'x_bigger_than': [], 'y_bigger_than': []}
+            for j in range(len(data)):
+                if i == j:
+                    continue
+                else:
+                    other_object_name = data[j]['object_name']
+                    for axis in ('x', 'y'):
+                        # Relative size is a discrete ('bigger_than') relation
+                        if data[i]['stroke_graph']['size'][axis] > \
+                                data[j]['stroke_graph']['size'][axis]:
+                            data[i]['relative_size'][axis+'_bigger_than'].append(other_object_name)
```

Comment:
It would be slightly simpler for `{axis}_bigger_than` to be a list of booleans. Then you don't need the `if`.

I think an f-string would make the relation name clearer: `f"{axis}_bigger_than"` vs. `axis+'_bigger_than_'`.